client.js

```diff
@@ -0,0 +1,427 @@
+        let index = 0;
+        for (const row of data.signalDataResponse.row) {
+          data.signalDataResponse.criterion[index] += this.timeDiff;
+          let signalNames = [];
+          for (const signalId of row.signalId) {
+            signalNames.push(this.idToName[signalId]);
+          }
+          const value = this._createValue(
+            signalNames,
+            row.minValues,
+            row.maxValues,
+            row.lastValues
+          );
+          dataPoints.push({
+            timestamp: data.signalDataResponse.criterion[index],
+            value: value
+          });
+          index++;
+        }
+        if (this.storedPromises[data.signalDataResponse.requestId]) {
+          const { resolve } = this.storedPromises[data.signalDataResponse.requestId];
+          delete this.storedPromises[data.signalDataResponse.requestId];
+          resolve(dataPoints);
+        }
+        break;
+      default:
+        console.error("Unknown message type", data.messageType);
+    }
+  }
+
+  _createValue(signalNames, minValues, maxValues, lastValues) {
+    const value = {};
+    for (let i = 0; i < signalNames.length; i++) {
+      value[signalNames[i]] = {
+        min: this._valueFromVariant(minValues[i]),
+        max: this._valueFromVariant(maxValues[i]),
+        last: this._valueFromVariant(lastValues[i])
+      };
+    }
+    return value;
+  }
+
+  _valueFromVariant(value) {
+    if (value.dValue !== undefined) return value.dValue;
+    if (value.fValue !== undefined) return value.fValue;
+    if (value.ui64Value !== undefined) return value.ui64Value;
+    if (value.i64Value !== undefined) return value.i64Value;
+    if (value.uiValue !== undefined) return value.uiValue;
+    if (value.iValue !== undefined) return value.iValue;
+    if (value.usValue !== undefined) return value.usValue;
+    if (value.sValue !== undefined) return value.sValue;
+    if (value.ucValue !== undefined) return value.ucValue;
+    if (value.cValue !== undefined) return value.cValue;
+    if (value.bValue !== undefined) return value.bValue;
+    if (value.strValue !== undefined) return value.strValue;
+    return null;
+  }
+
+  _sendQueuedRequests() {
+    for (const requestId in this.queuedRequests) {
+      const req = this.queuedRequests[requestId];
+      if (req === "logged_nodes") {
+        this._sendLoggedNodesRequest(requestId);
+      } else if (req === "log_limits") {
+        this._sendLogLimitsRequest(requestId);
+      } else if (Array.isArray(req) && req[0] === "node_values") {
+        this._reqDataPoints(req[1], req[2], req[3], req[4], requestId);
+      } else if (req === "api_version") {
+        this._sendApiVersionRequest(requestId);
+      }
+    }
+    this.queuedRequests = {};
+  }
+
+  _getRequestId() {
+    this.reqId += 1;
+    return this.reqId;
+  }
+
+  _timeRequest() {
+    if ((Date.now() / 1000) > this.lastTimeRequest + 10) {
+      this._updateTimeDiff();
+    }
+  }
+
+  _updateTimeDiff() {
```

Comment:
I think the Client should have some option to disable time sync. It has previously got in the way that sometimes people want raw database values from the logger without time adjustment